### PR TITLE
UIREC-175 Confirm if user would like to delete holdings record when d…

### PR DESCRIPTION
**Pull request by usavkov-epam**
…eleting piece

## Purpose

https://issues.folio.org/browse/UIREC-175
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/TitleDetails/hooks/useHoldingItems/useHoldingItems.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    ...kyOptions,
+    searchParams: {
+      limit: LIMIT_MAX,
+      ...kyOptions?.searchParams,
+      query: `holdingsRecordId==${holdingsRecordId}`,
+    },
+  };
+
+  const queryFn = () => ky.get(ITEMS_API, options).json();
+  const { data, isLoading, isFetching } = useQuery({ queryKey, queryFn, ...queryOptions });
```
Comment (NikitaSedyx) [inline src/TitleDetails/hooks/useHoldingItems/useHoldingItems.js:32]: plz disable query when holdingsRecordId is not defined, no reason to make a request when no holding connected

Comment (NikitaSedyx) [inline src/TitleDetails/hooks/useHoldingItems/useHoldingItems.js:32]: plz disable query when holdingsRecordId is not defined, no reason to make a request when no holding connected

File: src/TitleDetails/DeletePieceModal/DeletePieceModalContainer.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+const DeletePieceModalContainer = ({
+  onCancel,
+  onConfirm,
+  piece,
+  setIsLoading,
+}) => {
+  const { itemId, holdingsRecordId } = piece;
+  const { itemsCount, isFetching } = useHoldingItems(holdingsRecordId, { searchParams: { limit: 1 } });
+  const isLastConnectedItem = itemId && itemsCount === 1;
```
Comment (NikitaSedyx) [inline src/TitleDetails/DeletePieceModal/DeletePieceModalContainer.js:19]: what if we saved piece, created holding without item and decided to remove?

File: src/TitleDetails/hooks/useHoldingItems/useHoldingItems.js (hunk 1 above)
Comment (usavkov-epam) [reply src/TitleDetails/hooks/useHoldingItems/useHoldingItems.js]: updated

File: src/TitleDetails/DeletePieceModal/DeletePieceModalContainer.js (hunk 2 above)
Comment (usavkov-epam) [reply src/TitleDetails/DeletePieceModal/DeletePieceModalContainer.js]: As I understand it, if a piece is not related to an item, we show default confirm modal and delete only the piece.

Comment (NikitaSedyx) [reply src/TitleDetails/DeletePieceModal/DeletePieceModalContainer.js]: why not to remove holding? the idea is not to keep empty holdings

plz ask Dennis in the ticket

**Comment by NikitaSedyx** [comment]
plz fix code smells

File: src/TitleDetails/DeletePieceModal/DeletePieceModal.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      marginBottom0
+      onClick={onCancel}
+    >
+      <FormattedMessage id="ui-receiving.piece.actions.cancel" />
+    </Button>
+  );
+  const end = (
+    <>
+      {canDeleteHolding && (
+      <Button
```
Comment (NikitaSedyx) [inline src/TitleDetails/DeletePieceModal/DeletePieceModal.js:48]: indents plz

File: src/TitleDetails/hooks/useHoldingItems/useHoldingItems.js (hunk 1 above)
Comment (usavkov-epam) [reply src/TitleDetails/hooks/useHoldingItems/useHoldingItems.js]: updated

**Pull request merged**